planty/chats/views.py

```diff
@@ -0,0 +1,102 @@
+from django.shortcuts import redirect, render
+from .models import Message, Chat
+from .forms import MessageForm
+from django.utils import timezone
+from posts.models import Post
+from django.db.models import Q
+from django.contrib.auth.decorators import login_required
+from django.contrib import messages
+from django.core.exceptions import ObjectDoesNotExist
+from django.http import HttpResponseNotFound
+from .utils import return_list_last_messages_in_chat
+
+
+@login_required
+def first_message(request, pk):
+    try:
+        post_instance = Post.objects.get(id=pk)
+    except ObjectDoesNotExist:
+        return HttpResponseNotFound()
+
+    try:  # Chat already exists
+        chat = Chat.objects.get(Q(buyer=request.user) & Q(post=post_instance))
+        return redirect('chat', pk=chat.id)
+    except ObjectDoesNotExist:
+        form = MessageForm()
```

Comment:
Maybe you can move this line in an `else` of the `if request.method == "POST":` to avoid creating an unnecessary instance in some cases.